src/gui/editors/PianoRoll.cpp

```diff
@@ -844,6 +844,19 @@ void PianoRoll::loadMarkedSemiTones(const QDomElement & de)
 	m_markedSemiTones.erase(new_end, m_markedSemiTones.end());
 }
 
+void PianoRoll::setScrollbarPos(double posX)
+{
+	assert(posX >= 0.0 && posX < 1.0);
+
+	// Convert posX to position on scrollbar
+	auto scrollPosX = posX * (m_leftRightScroll->maximum() - m_leftRightScroll->minimum()) + m_leftRightScroll->minimum();
+
+	// Center the position within the editor
+	const auto tpp = 1.0 * TimePos::ticksPerBar() / m_ppb; // ticks per pixel
+	scrollPosX -= tpp * (0.5 * m_leftRightScroll->width() - m_topBottomScroll->width());
```

Comment:
I may be misunderstanding what you're trying to do here, but I think you want to subtract half the width of the part of the editor not covered by the vertical scrollbar, in which case that extra bit should be subtracted before halving. (And again, with the earlier suggestion applied, I don't think the vertical scrollbar width would need to be taken into account at all.)
```suggestion
	scrollPosX -= tpp * 0.5 * (m_leftRightScroll->width() - m_topBottomScroll->width());
```

Reply:
I tried doing it that way initially, but the scrollbar position became noticeably inaccurate at certain zoom levels and I didn't know why. The current kludgy way of doing it made it look better but didn't fix the problem completely. However, I think I just found out what was causing the inaccuracies and I'll push an actual fix soon.

Reply:
It should be fixed now. The scrollbar position is very accurate at each combination of Song Editor zoom level and Piano Roll zoom level that I tried.